lightstep/resource_opsgenie_destination.go

```diff
@@ -0,0 +1,128 @@
+package lightstep
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/lightstep/terraform-provider-lightstep/client"
+)
+
+func resourceOpsgenieDestination() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceOpsgenieDestinationCreate,
+		ReadContext:   resourceDestinationRead,
+		DeleteContext: resourceDestinationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceOpsgenieDestinationImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"project_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"destination_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the Opsgenie destination",
+			},
+			"url": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				Description:  "Opsgenie instance URL",
+				ValidateFunc: validation.IsURLWithScheme([]string{"https"}),
+			},
+			"auth": {
+				Type:        schema.TypeList,
+				MinItems:    1,
+				MaxItems:    1,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Basic auth used to authenticate with the Opsgenie instance",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
```

Comment:
could you add descriptions for these fields too? it's on my mind since I just went through and did it for a bunch of things